article.md

```diff
@@ -0,0 +1,612 @@
+
+```shell
+$ git clone git@github.com:ahmerb/ts-sqlize-code.git
+$ cd ts-sqlize-code
+$ git checkout 0.start
+```
+
+We can run the app with
+
+```shell
+$ yarn install
+$ yarn start
+```
+
+We now have a very simple Express app. Take a look at `app.ts`. It just defines a single request handler that returns the json `{ message: 'hello, world' }` and then tells Express to listen on port 3000.
+
+## Models, models, models
+
+Now, we want to setup Sequelize models that reflect the db schema we defined earlier. We will create a file for each model in the directory `src/models/`. Each file will export two types: `ModelAttributes` and `ModelInstance`. They'll also export a `ModelFactory` function which will actually set up the model and register it with Sequelize. At first, we're going to ignore associations. We'll deal with these later.
+
+Let's start with the `Users` model. Firstly, we'll define an interface `UserAttributes`. This interface defines all the attributes you specify when creating a new instance of the model. In our database schema, we defined that a `User` has `id`, `name`, `createdAt` and `updatedAt` attributes. Let's see how that looks.
+
+```typescript
+// src/models/User.ts
+export interface UserAttributes {
+  id?: number;
+  name: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+};
+```
+
+Simple. Note that `id`, `createdAt` and `updatedAt` are optional. This is because, when creating a new instance of a model, we don't want to be
+forced to specify these fields. Instead, we usually want Sequelize to automatically set them.
+
+The second type each model needs is a `ModelInstance`. This represents a Sequelize instance for an actual database row. For example, the function `User.find({ id: 1 })` returns an instance of the `User` model. It will have the `ModelAttributes` we defined (i.e., the table columns) as well as more Sequelize instance methods such as `.validate`, `.save`, `.update`, et cetera. For `Users`, the interface is defined as follows.
+
+```typescript
+// src/models/User.ts
+export interface UserInstance extends Sequelize.Instance<UserAttributes>, UserAttributes {
+  // At the moment, there's nothing more to add apart from the methods and attributes that the types `Sequelize.Instance<UserAttributes>` and `UserAttributes` give us. We'll add more here when we get on to adding associations.
+};
+```
+
+Now, we have the two types required. Next, we want to create a function that will actually define a Sequelize model for the `Users` table. The model should correspond to the `UserAttributes` and `UserInstance` types that we've already defined. We call this function `UserFactory`. In typical Sequelize manner, this function will take your Sequelize instance and a `DataTypes` object, and return the created Sequelize model. To define models in sequelize, we use the `sequelize.define<TInstance, TAttributes>()` function. The function takes an object which specifies the columns to create. 
```

Comment:
You could have some sort of reference as to why it's called a factory, e.g. "What we're doing here is creating a an instance without specifying its type, which some of you might recognize as [the factory pattern](https://en.wikipedia.org/wiki/Factory_method_pattern). Hence we'll call this function `UserFactory`."

Reply:
^ my wording might not be 100% accurate w.r.t to the definition, so you might want to double check that 😄